src/python/pants/java/junit/junit_xml_parser.py

```diff
@@ -170,3 +170,56 @@ def parse_junit_xml_file(path):
     parse_junit_xml_file(junit_xml_path)
 
   return dict(failed_targets)
+
+
+def parse_test_info(junit_xml_path, error_handler):
+  """
+  Parses the junit file for info needed about each test
+  info needed:
+    - result code
+    - run time duration
+    - name
+  :param string junit_xml_path:
+  :param error_handler:
+  :return: dict of tests
+  """
+  tests_in_path = {}
+
+  def parse_junit_xml_file(path):
+    SUCCESS = 0
+    SKIPPED = 5000
+    FAILURE = 10000
+    ERROR = 15000
+
+    try:
+      xml = XmlParser.from_file(path)
+      for testcase in xml.parsed.getElementsByTagName('testcase'):
+        test_info = {'time': float(testcase.getAttribute('time'))}
+        test_info.update({'classname': testcase.getAttribute('classname')})
+
+        test_error = testcase.getElementsByTagName('error')
+        test_fail = testcase.getElementsByTagName('failure')
+        test_skip = testcase.getElementsByTagName('skipped')
+
```

Comment:
There is also the option of adding all the xml parsing to a util file since there is also some overlap with parsing the failed tests' targets in [pytest_run.py](https://github.com/pantsbuild/pants/blob/ae1ea88d5a280120dff3a4cff2141b774ccca139/src/python/pants/backend/python/tasks2/pytest_run.py#L438) and [junit_run.py](https://github.com/pantsbuild/pants/blob/ae1ea88d5a280120dff3a4cff2141b774ccca139/src/python/pants/java/junit/junit_xml_parser.py#L148)